packages/model-viewer/src/features/annotation.ts

```diff
@@ -0,0 +1,245 @@
+ */
+export class Hotspot extends CSS2DObject {
+  public normal: Vector3;
+  private referenceCount: number;
+
+  constructor(config: HotspotConfiguration) {
+    const wrapper = document.createElement('div');
+    const slot = document.createElement('slot');
+    slot.name = config.name;
+    wrapper.appendChild(slot);
+    super(wrapper);
+    this.normal = new Vector3(0, 1, 0);
+    this.updatePosition(config.position);
+    this.updateNormal(config.normal);
+    this.referenceCount = 1;
+  }
+
+  increment() {
+    ++this.referenceCount;
+  }
+
+  decrement(): boolean {
+    return --this.referenceCount <= 0;
+  }
+
+  updatePosition(position?: string) {
+    if (position == null)
+      return;
+    const positionNodes = parseExpressions(position)[0].terms;
+    for (let i = 0; i < 3; ++i) {
+      this.position.setComponent(
+          i, normalizeUnit(positionNodes[i] as NumberNode<'m'>).number);
+    }
+  }
+
+  updateNormal(normal?: string) {
+    if (normal == null)
+      return;
+    const normalNodes = parseExpressions(normal)[0].terms;
+    for (let i = 0; i < 3; ++i) {
+      this.normal.setComponent(
+          i, normalizeUnit(normalNodes[i] as NumberNode<'m'>).number);
+    }
+  }
+}
+
+export declare interface AnnotationInterface {
+  updateHotspot(config: HotspotConfiguration): void;
+}
+
+/**
+ * AnnotationMixin implements a declarative API to add hotspots and annotations.
+ * Child elements of the <model-viewer> element that have a slot name that
+ * begins with "hotspot" and data-position and data-normal attributes in
+ * the format of the camera-target attribute will be added to the scene and
+ * track the specified model coordinates.
+ */
+export const AnnotationMixin = <T extends Constructor<ModelViewerElementBase>>(
+    ModelViewerElement: T): Constructor<AnnotationInterface>&T => {
+  class AnnotationModelViewerElement extends ModelViewerElement {
+    private[$annotationRenderer] = new CSS2DRenderer();
+    private[$hotspotMap] = new Map<string, Hotspot>();
+    private[$mutationCallback] = (mutations: Array<unknown>) => {
+      mutations.forEach((mutation) => {
+        // NOTE: Be wary that in ShadyDOM cases, the MutationRecord
+        // only has addedNodes and removedNodes (and no other details).
+        if (!(mutation instanceof MutationRecord) ||
+            mutation.type === 'childList') {
+          (mutation as MutationRecord).addedNodes.forEach((node) => {
+            this[$addHotspot](node);
+          });
+          (mutation as MutationRecord).removedNodes.forEach((node) => {
+            this[$removeHotspot](node);
+          });
+        }
+      });
+    };
+    private[$observer] = new MutationObserver(this[$mutationCallback]);
+
+    connectedCallback() {
+      super.connectedCallback();
+      const {domElement} = this[$annotationRenderer];
+      const {style} = domElement;
+      style.pointerEvents = 'none';
+      style.position = 'absolute';
+      style.top = '0';
```

Comment:
Let's just add a class to this element (like `domElement.classList.add('annotation-container')`) and put the styles for it in `template.ts`. We can do something similar for each slot wrapper.